src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -267,64 +299,72 @@ func (c controller) unmarshalStatusAnnotation(content []byte) (*metal3v1alpha1.B
 	return bmhStatus, nil
 }
 
-func (c controller) unpatchEtcd() {
+func (c controller) unpatchEtcd() bool {
 	c.log.Infof("Unpatching etcd")
-	for {
-		if err := c.kc.UnPatchEtcd(); err != nil {
-			c.log.Error(err)
-			continue
-		}
-		break
+	if err := c.kc.UnPatchEtcd(); err != nil {
+		c.log.Error(err)
+		return false
 	}
-
+	return true
 }
 
 // AddRouterCAToClusterCA adds router CA to cluster CA in kubeconfig
-func (c controller) addRouterCAToClusterCA() {
+func (c controller) addRouterCAToClusterCA() bool {
 	cmName := "default-ingress-cert"
 	cmNamespace := "openshift-config-managed"
 	c.log.Infof("Start adding ingress ca to cluster")
-	for {
-		caConfigMap, err := c.kc.GetConfigMap(cmNamespace, cmName)
+	caConfigMap, err := c.kc.GetConfigMap(cmNamespace, cmName)
 
-		if err != nil {
-			c.log.WithError(err).Errorf("fetching %s configmap from %s namespace", cmName, cmNamespace)
-			continue
-		}
+	if err != nil {
+		c.log.WithError(err).Errorf("fetching %s configmap from %s namespace", cmName, cmNamespace)
+		return false
+	}
 
-		c.log.Infof("Sending ingress certificate to inventory service. Certificate data %s", caConfigMap.Data["ca-bundle.crt"])
-		err = c.ic.UploadIngressCa(caConfigMap.Data["ca-bundle.crt"], c.ClusterID)
-		if err != nil {
-			c.log.WithError(err).Errorf("Failed to upload ingress ca to assisted-service")
-			continue
-		}
-		c.log.Infof("Ingress ca successfully sent to inventory")
-		return
+	c.log.Infof("Sending ingress certificate to inventory service. Certificate data %s", caConfigMap.Data["ca-bundle.crt"])
+	err = c.ic.UploadIngressCa(caConfigMap.Data["ca-bundle.crt"], c.ClusterID)
+	if err != nil {
+		c.log.WithError(err).Errorf("Failed to upload ingress ca to assisted-service")
+		return false
 	}
-}
+	c.log.Infof("Ingress ca successfully sent to inventory")
+	return true
 
-func (c controller) waitForConsole() {
-	c.log.Infof("Waiting for console pod")
+}
 
-	// TODO maybe need some timeout?
-	for {
-		pods, err := c.kc.GetPods("openshift-console", map[string]string{"app": "console", "component": "ui"})
-		if err != nil {
-			c.log.WithError(err).Warnf("Failed to get console pods")
-			continue
+func (c controller) validateConsolePod() bool {
+	c.log.Infof("Checking if console pod is running")
+	pods, err := c.kc.GetPods("openshift-console", map[string]string{"app": "console", "component": "ui"})
+	if err != nil {
+		c.log.WithError(err).Warnf("Failed to get console pods")
+		return false
+	}
+	for _, pod := range pods {
+		if pod.Status.Phase == "Running" {
+			c.log.Infof("Found running console pod")
+			return true
 		}
-		for _, pod := range pods {
-			if pod.Status.Phase == "Running" {
-				c.log.Infof("Found running console pod")
-				return
-			}
-			c.log.Infof("Console pod is in status %s. Continue waiting for it", pod.Status.Phase)
+		c.log.Infof("Console pod is in status %s. Continue waiting for it", pod.Status.Phase)
+	}
+	return false
+}
+
+func (c controller) validateClusterVersion() bool {
```

Comment:
Make sure this works and come back with OperatorAvailable==True within the timeout.
Note that this might effect our success rate, any idea if the openshift installer check this as well?

Reply:
sure it is, thats why i added this